### PR TITLE
Fixed sonnet json formatting issue

### LINKED ISSUE #292: Parsing JSON with newlines
If the model response looks like:

```
{
  "summary": "The excerpt discusses different approaches to representing protein structures for machine learning, including equivariant methods. It mentions that scalar quantities of proteins like interaction interfaces are intrinsically rotation and translation invariant. The text describes several equivariant approaches:

1. Tensor Field Networks (TFN) using rotation equivariant convolutions based on spherical harmonics
2. Cormorant networks improving on TFN with Clebsch-Gordan non-linearity 
3. Geometric vector perceptron (GVP) using linear operations on vectors respecting rotation equivariance
4. Equivariant graph neural networks

The excerpt notes that TFN architectures have been successful in predicting protein-protein complex quality. However, it does not explicitly state that equivariant layers are definitively better at modeling protein structure compared to other approaches.",
  "relevance_score": 7
}
```

we do not parse it correctly. We should be able to handle that.

### PR DESCRIPTION
**Pull request by whitead**
Fixes #292

### CODE REVIEW

File: paperqa/utils.py [hunk 1]
```diff
@@ -184,4 +184,12 @@ def llm_read_json(text: str) -> dict:
     text = "{" + text.split("{", 1)[-1]
     # split anything after the last }
     text = text.rsplit("}", 1)[0] + "}"
+
+    # escape new lines within strings
+    def replace_newlines(match):
+        return match.group(0).replace("\n", "\\n")
+
+    pattern = r'"(?:[^"\\]|\\.)*"'
+    text = re.sub(pattern, replace_newlines, text)
```
Comment (jamesbraza) [inline paperqa/utils.py:193]: TIL `repl` arg can be a function

File: paperqa/utils.py [hunk 2]
```diff
@@ -184,4 +184,12 @@ def llm_read_json(text: str) -> dict:
     text = "{" + text.split("{", 1)[-1]
     # split anything after the last }
     text = text.rsplit("}", 1)[0] + "}"
+
+    # escape new lines within strings
+    def replace_newlines(match):
+        return match.group(0).replace("\n", "\\n")
+
+    pattern = r'"(?:[^"\\]|\\.)*"'
```
Comment (jamesbraza) [inline paperqa/utils.py:192]: Bonus points for regex 101 link

File: paperqa/utils.py [hunk 3]
```diff
@@ -184,4 +184,12 @@ def llm_read_json(text: str) -> dict:
     text = "{" + text.split("{", 1)[-1]
     # split anything after the last }
     text = text.rsplit("}", 1)[0] + "}"
+
+    # escape new lines within strings
+    def replace_newlines(match):
+        return match.group(0).replace("\n", "\\n")
```
Comment (jamesbraza) [inline paperqa/utils.py:190]: Is there a latent `NoneType` error here?

File: paperqa/utils.py [hunk 4]
```diff
@@ -184,4 +184,12 @@ def llm_read_json(text: str) -> dict:
     text = "{" + text.split("{", 1)[-1]
     # split anything after the last }
     text = text.rsplit("}", 1)[0] + "}"
+
+    # escape new lines within strings
```
Comment (jamesbraza) [inline paperqa/utils.py:188]: What do you think of [`re.escape`](https://docs.python.org/3/library/re.html#re.escape)?

File: tests/test_paperqa.py [hunk 5]
```diff
@@ -457,6 +458,22 @@ def test_llm_read_json(example: str):
     assert llm_read_json(example) == {"example": "json"}
 
 
+def test_llm_read_json_newlines():
```
Comment (jamesbraza) [inline tests/test_paperqa.py:461]: Can you add a docstring or comment that newlines are an edge case that broke us

**Review by jamesbraza — APPROVED** [review]
LGTM, added some comments

File: paperqa/utils.py (hunk 3 above)
Comment (whitead) [reply paperqa/utils.py:190]: Added a type - I don't think sub will call on None matches

File: tests/test_paperqa.py (hunk 5 above)
Comment (whitead) [reply tests/test_paperqa.py:461]: Thanks! Done

File: paperqa/utils.py (hunk 4 above)
Comment (whitead) [reply paperqa/utils.py:188]: Don't see the connection

File: paperqa/utils.py (hunk 2 above)
Comment (whitead) [reply paperqa/utils.py:195]: Added!
